.github/workflows/build-docker-with-wheels-artifact.yaml

```diff
@@ -0,0 +1,84 @@
+name: "Build Docker Release Image artifact"
+
+on:
+  workflow_call:
+    inputs:
+      arch:
+        required: false
+        type: string
+        default: '["grayskull", "wormhole_b0"]'
+      build-type:
+        required: false
+        type: string
+        default: "Release"
+      tracy:
+        required: false
+        type: boolean
+        default: false
+        description: "Build with tracy enabled"
+  workflow_dispatch:
+    inputs:
+      arch:
+        required: false
+        type: string
+        default: '["grayskull", "wormhole_b0"]'
+      build-type:
+        required: false
+        type: choice
+        options:
+          - Release
+          - Debug
+          - RelWithDebInfo
+          - CI
+      tracy:
+        required: false
+        type: boolean
+        default: false
+        description: "Build with tracy enabled"
+jobs:
+  build-docker-images:
+    strategy:
+      matrix:
+        arch: ${{ fromJson(inputs.arch || '["grayskull", "wormhole_b0", "blackhole"]') }}
+        os: [ ubuntu-20.04 ]
+        docker_image_tag: [ tt-metalium/ubuntu-20.04-amd64 ]
+    env:
+      TT_METAL_ENV: ${{ vars.TT_METAL_ENV }}
+      ARCH_NAME: ${{ matrix.arch }}
+      IMAGE_TAG: dev-${GITHUB_ACTOR}.${{ matrix.arch }}
+      TT_METAL_DOCKER_IMAGE: tt-metalium/ubuntu-20.04-amd64
+      TT_METAL_DOCKER_IMAGE_TAG: ${{ github.sha }}
+    environment: dev
+    runs-on: build-docker
+    steps:
+      - uses: tenstorrent-metal/metal-workflows/.github/actions/checkout-with-submodule-lfs@v2.0.0
+      - uses: actions/download-artifact@v4
+        with:
+          name: eager-dist-${{ matrix.os }}-${{ matrix.arch }}
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: https://ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Determine Docker Image Tag
+        id: set-image
+        run: |
+          if [[ "${GITHUB_REF_NAME}" == "main" ]]; then
+            echo "IMAGE_TAG=latest" >> $GITHUB_ENV
+          else
+            echo "IMAGE_TAG=dev-${GITHUB_ACTOR}.${{ matrix.arch }}.${{ github.sha }}" >> $GITHUB_ENV
+          fi
+      - name: Build Docker image
+        run: |
+          ARCH_NAME=${{ matrix.arch}}
+          # Do we not have to unzip? 
+          # unzip eager-dist-${{ matrix.os }}-${{ matrix.arch }}.zip
+          WHEEL_FILENAME=$(ls -1 *.whl)
+          echo $WHEEL_FILENAME
+          docker build -f dockerfile/tt-metalium.Dockerfile -t ${{ env.TT_METAL_DOCKER_IMAGE_TAG }}-${{ matrix.arch }} . --build-arg="WHEEL_FILENAME=${WHEEL_FILENAME}" --build-arg="IMAGE_TAG=dev-${GITHUB_ACTOR}"
```

Comment:
Create a separate workflow to create a release image and run tests on it.

Reply:
Create an issue

Reply:
https://github.com/tenstorrent/tt-metal/issues/11973

Reply:
I would personally keep this file on another branch rather than this PR, and work on it on the side for release images for that issue.

Just to separate concerns.